gensim/models/fasttext.py

```diff
@@ -0,0 +1,363 @@
+            new_count = 0
+            for i, ngram in enumerate(new_ngrams):
+                ngram_hash = Ft_Wrapper.ft_hash(ngram)
+                if ngram_hash not in self.wv.hash2index:
+                    self.wv.hash2index[ngram_hash] = new_count + len(self.wv.vocab) + self.old_hash2index_len
+                    self.wv.ngrams[ngram] = self.wv.hash2index[ngram_hash]
+                else:
+                    self.wv.ngrams[ngram] = self.wv.hash2index[ngram_hash]
+
+            old_vocab_rows = self.wv.syn0_all[0:self.old_vocab_len, ]
+            old_ngram_rows = self.wv.syn0_all[self.old_vocab_len:, ]
+
+            rand_obj = np.random
+            rand_obj.seed(self.seed)
+            new_vocab_rows = rand_obj.uniform(-1.0 / self.vector_size, 1.0 / self.vector_size, (len(self.wv.vocab) - self.old_vocab_len, self.vector_size))
+            new_ngram_rows = rand_obj.uniform(-1.0 / self.vector_size, 1.0 / self.vector_size, (len(self.wv.hash2index) - self.old_hash2index_len, self.vector_size))
+
+            self.wv.syn0_all = vstack([old_vocab_rows, new_vocab_rows, old_ngram_rows, new_ngram_rows])
+
+    def reset_ngram_weights(self):
+        rand_obj = np.random
+        rand_obj.seed(self.seed)
+        for index in range(len(self.wv.vocab) + len(self.wv.ngrams)):
+            self.wv.syn0_all[index] = rand_obj.uniform(-1.0 / self.vector_size, 1.0 / self.vector_size, self.vector_size)
+
+    def _do_train_job(self, sentences, alpha, inits):
+        work, neu1 = inits
+        tally = 0
+        if self.sg:
+            tally += train_batch_sg(self, sentences, alpha, work)
+        else:
+            tally += train_batch_cbow(self, sentences, alpha, work, neu1)
+
+        return tally, self._raw_word_count(sentences)
+
+    def train(self, sentences, total_examples=None, total_words=None,
+              epochs=None, start_alpha=None, end_alpha=None,
+              word_count=0, queue_factor=2, report_delay=1.0):
+        self.neg_labels = []
+        if self.negative > 0:
+                # precompute negative labels optimization for pure-python training
+                self.neg_labels = zeros(self.negative + 1)
+                self.neg_labels[0] = 1.
+
+        Word2Vec.train(self, sentences, total_examples=self.corpus_count, epochs=self.iter,
+            start_alpha=self.alpha, end_alpha=self.min_alpha)
+        self.get_vocab_word_vecs()
+
+    def __getitem__(self, word):
+        return self.word_vec(word)
+
+    def get_vocab_word_vecs(self):
+        for w, v in self.wv.vocab.items():
+            word_vec = self.wv.syn0_all[v.index]
+            ngrams = self.wv.ngrams_word[w]
+            ngram_weights = self.wv.syn0_all
+            for ngram in ngrams:
+                word_vec += ngram_weights[self.wv.ngrams[ngram]]
+            word_vec /= (len(ngrams) + 1)
+            self.wv.syn0[v.index] = word_vec
+
+    def word_vec(self, word, use_norm=False):
```

Comment:
Any reason for not reusing `word_vec` from `FastTextKeyedVectors` here?

Reply:
Done

Reply:
I get this error if I try to access an OOV word:

`print model["use"] # only "user" is available in the vocabulary`

         86         else:
         87             word_vec = np.zeros(self.syn0_ngrams.shape[1])
    ---> 88             ngrams = compute_ngrams(word, self.min_n, self.max_n)
         89             ngrams = [ng for ng in ngrams if ng in self.ngrams]
         90             if use_norm:

    AttributeError: 'FastTextKeyedVectors' object has no attribute 'min_n'

Looks like `FastTextKeyedVectors` class has no `min_n` attribute?

Reply:
Here's the minimal script to reproduce my issue https://gist.github.com/rilut/31f41d5cf3da075d43cf7e4f2c993b76

Thanks 🙏

Reply:
Btw, since `self.wv = FastTextKeyedVectors()` why not `self.wv.word_vec(word, word_norm)`? (just asking)

Reply:
@rilut Seems like I missed setting `min_n` and `max_n` in `FastTextKeyedVectors` after I had refactored the code b/w the wrapper and main `models.FastText` class.
Anyway, thanks a lot for pointing out this! I have made the necessary changes for this now and also added a unittest to ensure that this error doesn't go unnoticed in the future.
And yes, `FastTextKeyedVectors.word_vec(self.wv, word, use_norm=use_norm)` and `self.wv.word_vec(word, word_norm)` are equivalent. I preferred the former form at the time since it made the borrowed usage of the function more explicit.

Reply:
@chinmayapancholi13 oh ok that's good. Thanks for your fix and hard work! I really appreciated it.